docs/6.x/storage.md

```diff
@@ -0,0 +1,346 @@
+
+```yaml
+storage:
+  openebs:
+    enabled: true
+```
+
+When OpenEBS is enabled, it will be packaged in the cluster image tarball
+alongside other dependencies during the `tele build` process. During the
+cluster installation, OpenEBS operator and other components will be installed
+in the `openebs` namespace.
+
+!!! note "Privileged containers"
+    Certain OpenEBS components need to operate in a privileged mode so
+    privileged containers will be implicitly allowed when integration with
+    OpenEBS is enabled. See [their FAQ](https://docs.openebs.io/docs/next/faq.html#why-ndm-priviledged)
+    for details.
+
+### Enable OpenEBS During Upgrade
+
+OpenEBS can be enabled for existing Gravity clusters when they are upgraded
+to a new version that has OpenEBS integration turned on.
+
+To enable it in the existing cluster:
+
+* Update your cluster image manifest to enable OpenEBS integration like shown above.
+* Build a new version of the cluster image using `tele build`. See [Building a Cluster Image](/pack/#building-a-cluster-image) for details.
+* Upgrade the existing cluster to this new version. See [Upgrading a Cluster](/cluster/#updating-a-cluster) for details.
+
+OpenEBS will be installed and configured during the upgrade operation.
+
+## Configure OpenEBS
+
+OpenEBS scans the cluster nodes to discover block devices attached to the nodes
+and makes them available for use via persistent volumes. In the default configuration
+it takes into account all devices it finds, excluding some well-known system
+ones such as loop/devicemapper devices, RAM disks and so on. Sometimes it may
+be necessary to only consider certain devices on the nodes, for example to
+exclude some disks that are attached to the node but should not be allocated
+for the in-cluster persistent storage use.
+
+Gravity provides a way to configure OpenEBS devices via a `PersistentStorage`
+resource:
+
+```yaml
+kind: PersistentStorage
+version: v1
+spec:
+  # OpenEBS-specific configurations.
+  openebs:
+    # Filters for OpenEBS block device discovery.
+    filters:
+      # Include/exclude specific devices or device patterns.
+      devices:
+        include: []
+        exclude: ["/dev/sdb", "/dev/sdc"]
+      # Exclude devices mounted under specific paths.
+      mountPoints:
+        exclude: ["/", "/etc/hosts", "/boot"]
+      # Include/exclude devices from specific vendors.
+      vendors:
+        include: []
+        exclude: ["CLOUDBYT", "OpenEBS"]
+```
+
+!!! note "Default filters"
+    Default OpenEBS filters will be merged with the filters configured via
+    a `PersistentStorage` resource so you don't need to add them explicitly.
+
+Suppose that we want to include only two specific devices, `/dev/sdd` and
+`/dev/sde`, in the OpenEBS pool. Let's define the following resource in a
+`storage.yaml` file:
+
+```yaml
+kind: PersistentStorage
+version: v1
+spec:
+  openebs:
+    filters:
+      devices:
+        include: ["/dev/sdd", "/dev/sde"]
```

Comment:
I'm not sure I follow this. The above says the filters get merged with the default filters. 

So if unset, it will include all devices but the defaults to be denied?
Then if set, it will whitelist only those explicitly listed, with an implicit deny all?

Reply:
Yes, if `include` is present, then only matching devices will be considered. If it's empty, then all devices will be considered, except those matching `exclude` filter. I think I'll add this explanation here and also link back to the OpenEBS docs.

Reply:
I've expanded the docs on this a bit, let me know if that's more clear now.